### DISCUSSION TITLE
chigger example for reading in exodus mesh info

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#21013] [Q&A Tools] [answered]
Is there an example/test for chigger that gets the nodal coordinates from an exodus mesh and/or variable values (nodal or elemental) based on coordinates? Note I'm thinking about a mechanics problem so I'd be interested in the coordinates on the *displaced* mesh.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Looking at the tests in chigger/tests/displacement, the way to deal with the displaced mesh is to output the displaced exodus directly and read this. 

For nodal coordinates and variable values I could not find anything in the tests. I think Chigger is more of a plotting tool than a python analysis of exodus files tool personally. But others may disagree. 
Anyway
in python/chigger/exodus/ExodusSourceLineSampler.py you see code for a probing a variable at a desired location. I think this should do what you need. 

Guillaume

**Reply by GiudGiud** [reply]
if your final goal is plotting and not probing the file value at a given point then maybe there's a better test. 
Tagging @aeslaughter as he remains the expert on this.

**Reply by jessecarterMOOSE** [reply]
My interest really isn't plotting at this point. I'd like to do more analysis of the mesh after the fact rather than setting up nodesets and sidesets and postprocessors if it's less effort.

**Reply by GiudGiud** [reply]
One way I do this is to load the results exodus file and run all the postprocessing I need with Problem/solve=false
